src/components/POLine/Location/LocationForm.js

```diff
@@ -23,13 +23,21 @@ const reduceLocations = (locations, propName) => {
   return locations.reduce(reducer, 0);
 };
 
+const validateNotNegative = (value) => {
+  const parsedValue = parseInt(value, 10);
+
+  return Number.isNaN(parsedValue) || parsedValue < 0
```

Comment:
I'm not sure that we should forbid such falsy values as `''`, `undefined` or `null`. There are no clue if the field is required or not.

Reply:
but if we expect some number more than -1, it means for me we should prevent setting undefined, null and empty string isn't it?

Reply:
No, you're wrong, user might want to clear the input and it will be not valid per your logic

Reply:
just stick to atomic validator you've suggested before - the should wait `Number` type value and check simple conditions, just to make possible to stack them in any order you want. So for `required` - a separate validator, and it's not the case for locations' quantities fields.

Reply:
let's look at the case we want validate for not negative value only, and this case without any additional validation for imputed value like '' and etc. we won't give a warning, because are waiting only Number. What should we do this case? And  related with the current piece of code I've added conditional validation according to the order format.